### PR TITLE
Add a note saying that `{u8,i8}::from_{be,le,ne}_bytes` is meaningless

### PR DESCRIPTION
**Pull request by tbu-**

### CODE REVIEW

File: library/core/src/num/mod.rs [hunk 1]
```diff
@@ -77,6 +77,31 @@ pub use saturating::Saturating;
...
+    };
+}
+
+macro_rules! i8_xe_bytes_doc {
+    () => {
+        "
+
+**Note**: This function is technically useless on `i8`. Byte order does not
+exist as a concept for byte-sized integers. This function is only provided in
+symmetry with larger integer types. You can cast from and two `u8` using `as
```
Comment (cyrgani) [inline library/core/src/num/mod.rs:98]: two -> to

Comment (tbu-) [reply library/core/src/num/mod.rs]: Fixed.

**Comment by jhpratt** [comment]
I think saying "no-op" would be better than "useless". It _does_ have a purpose, after all (primarily macros).

**Comment by tbu-** [comment]
> I think saying "no-op" would be better than "useless". It _does_ have a purpose, after all (primarily macros).

It's not really a no-op, it's taking the byte and putting it into an array. The function name (or documentation or doctest) doesn't really make sense for `u8` as a `u8` doesn't have a concept of byte order (since it only has a single byte, "byte order" doesn't make sense).

I doubt the existence of them was even thought about when initially introducing them. Looking at the git blame, it was apparently me who added them. :D https://github.com/rust-lang/rust/pull/51919. I think it's only an accident that they exist on `u8`/`i8` at all.

I'm open for another word that describes that the function's names already convey nonsensical behavior, but I think "no-op" is not it.

**Comment by jhpratt** [comment]
> It's not really a no-op, it's taking the byte and putting it into an array. 

True, I spoke too quickly. In any situation, "useless" probably isn't the right word here. I don't have any concrete alternatives off-hand.

**Comment by tbu-** [comment]
Changed to "meaningless".

**Comment by jhpratt** [comment]
Might want to double check, as it doesn't look like that actually happened :slightly_smiling_face:

**Comment by tbu-** [comment]
You're right, I forgot to `git add` the changes.

@bors r=@jhpratt

**Pull request merged**